test/reline/test_config.rb

```diff
@@ -275,6 +275,46 @@ def test_unmatched_endif
     assert_equal "INPUTRC:1: unmatched endif", e.message
   end
 
+  def test_if_with_mode
+    @config.read_lines(<<~LINES.lines)
+      $if mode=vi
+        "\C-e": history-search-backward # comment
+      $else
+        "\C-f": history-search-forward
+      $endif
+    LINES
+
+    assert_equal({[5] => :history_search_backward}, @config.instance_variable_get(:@additional_key_bindings)[:vi_insert])
+    assert_equal({}, @config.instance_variable_get(:@additional_key_bindings)[:vi_command])
+    assert_equal({}, @config.instance_variable_get(:@additional_key_bindings)[:emacs])
```

Comment:
I think this test case is wrong.

With this inputrc file saved to `./inputrc`
```
$if mode=vi
  "x": "X"
$else
  "y": "Y"
$endif
```
Only `"y"→"Y"`, the `$else` part is reflected.
```
$ echo x | INPUTRC=./inputrc ruby -rreadline -e "p Readline::VERSION; Readline.readline'>'"
"8.2"
>x
$ echo y | INPUTRC=./inputrc ruby -rreadline -e "p Readline::VERSION; Readline.readline'>'"
"8.2"
>Y
```

Reply:
Thanks for letting me know! I've fixed it and added tests.